// ##########################################################
// # Сервис позволяет верифицировать денежные операции      #
// ##########################################################

syntax = "proto3";
package user_profile;

message UserDepositRequest {
    string name = 1;
    string password_hash = 2;
    double amount = 3;
    string user_uuid = 4;
}

message StatusResponse {
    enum StatusCode {
        OK = 0;
        USER_NOT_FOUND = 1;
        ERROR = 2;
    }

    StatusCode code = 1;
    string message = 2;
}

service MoneyTransacton {
    rpc deposit(UserDepositRequest) returns (StatusResponse);
    rpc withdraw(UserDepositRequest) returns (StatusResponse);
}